### ISSUE TITLE
Is it a good idea to export the encode and decoder of the generated types?

### ISSUE THREAD

**Issue by choonkeat** [harmboschloo/graphql-to-elm#12]
For pagination and deep linking purposes, I’ll need to persist `Variables` as a url query. This means I need to encode and decode it client side. 

What do you think?

**Comment by harmboschloo** [comment]
Yeah, sound like a good idea. Adding `encodeVariables` would be easy. For adding `decodeVariables` I'll have to have a look. 

Another option for `decodeVariables` would be to treat the variables as a json `Value` and expose the response decoder and query string. Then you can create your own `Operation`.

**Comment by choonkeat** [comment]
Exposing `Variables` as an opaque json `Value` means I could persist and pass it around as url parameters.

However I won’t be able to update a current value with before/after page cursors, e.g. `{ a | pageNumber = a.pageNumber + 1 }`

I took a look at `generateDecoders` but couldn’t quite follow. I’ll take a look again and PR if possible. Pointers appreciated 🙇‍♂️

**Comment by harmboschloo** [comment]
> However I won’t be able to update a current value with before/after page cursors, e.g. { a | pageNumber = a.pageNumber + 1 }

Good point.

> I took a look at generateDecoders but couldn’t quite follow. I’ll take a look again and PR if possible. Pointers appreciated 🙇‍♂️

This might be a good point to start: https://github.com/harmboschloo/graphql-to-elm/blob/master/src/gen/queries/elmIntel.ts#L173

It's where the decoder data for the output is being created. We may be able to do the same for the input (variables, see a few lines above).
